### PR TITLE
Add color interpolation

### PR DESCRIPTION
**Pull request by karol-bisztyga**
## Description

Add color interpolation, taken from https://github.com/wcandillon/react-native-redash/blob/master/src/Colors.ts

### CODE REVIEW

**Comment by DmitryFt** [comment]
Your solution is not going for Android

**Comment by karol-bisztyga** [comment]
Just built it locally and it worked fine. I don't think it's possible that this PR ruins build like that as changes are only made in javascript.
Also current alpha is 7.
Jakub's got a point, try to rebuild the project.

File: src/reanimated2/Colors.js [hunk 1]
```diff
@@ -420,3 +423,200 @@ export default function processColor(color) {
...
+  };
+};
+
+export const hsv2color = (h, s, v) => {
+  'worklet';
+  const { r, g, b } = hsv2rgb(h, s, v);
+  return rgbaColor(r, g, b);
+};
+
+export const colorForBackground = (r, g, b) => {
```
Comment (Szymon20000) [inline src/reanimated2/Colors.js:518]: Why do we need this?

**Review by Szymon20000 — APPROVED** [review]
Looks fine! Left only one question.

**Comment by nhannah** [comment]
Can anyone confirm they have the alpha channel working with this interpolateColor function? Without alpha I have no issue, but adding an alpha I get a rainbow of colors depending on the value. @wcandillon Have you had this issue? I see it came from your redash repo originally.

**Comment by jakub-gonet** [comment]
@nhannah, could you open a new issue with some repro so we can verify it?

**Pull request merged**